dspace-api/src/test/java/org/dspace/content/RelationshipServiceImplTest.java

```diff
@@ -0,0 +1,287 @@
+                testRel,0,0));
+
+        context.restoreAuthSystemState();
+    }
+
+    @Test
+    public void testDelete() throws Exception {
+        context.turnOffAuthorisationSystem();
+        when(authorizeService.isAdmin(context)).thenReturn(true);
+        MetadataValue metVal = mock(MetadataValue.class);
+        List<MetadataValue> metsList = new ArrayList<>();
+        List<Relationship> leftTypelist = new ArrayList<>();
+        List<Relationship> rightTypelist = new ArrayList<>();
+        Item leftItem = mock(Item.class);
+        Item rightItem = mock(Item.class);
+        RelationshipType testRel = new RelationshipType();
+        EntityType leftEntityType = mock(EntityType.class);
+        EntityType rightEntityType = mock(EntityType.class);
+        testRel.setLeftType(leftEntityType);
+        testRel.setRightType(rightEntityType);
+        testRel.setLeftLabel("Entitylabel");
+        testRel.setRightLabel("Entitylabel");
+        testRel.setLeftMinCardinality(0);
+        testRel.setRightMinCardinality(0);
+        metsList.add(metVal);
+        relationship = getRelationship(leftItem, rightItem, testRel, 0,0);
+        leftTypelist.add(relationship);
+        rightTypelist.add(relationship);
+        when(virtualMetadataPopulator.isUseForPlaceTrueForRelationshipType(relationship.getRelationshipType(), true))
+                .thenReturn(true);
+        when(authorizeService.authorizeActionBoolean(context, relationship.getLeftItem(), Constants.WRITE))
+                .thenReturn(true);
+        when(authorizeService.authorizeActionBoolean(context, relationship.getRightItem(), Constants.WRITE))
+                .thenReturn(true);
+        when(relationshipService.findByItem(context,leftItem)).thenReturn(leftTypelist);
+        when(relationshipService.findByItem(context,rightItem)).thenReturn(rightTypelist);
+        when(leftEntityType.getLabel()).thenReturn("Entitylabel");
+        when(rightEntityType.getLabel()).thenReturn("Entitylabel");
+        when(metVal.getValue()).thenReturn("Entitylabel");
+        when(metsList.get(0).getValue()).thenReturn("Entitylabel");
+        when(relationshipService.findByItemAndRelationshipType(context, leftItem, testRel, true))
+                .thenReturn(leftTypelist);
+        when(itemService.getMetadata(leftItem, "relationship", "type", null, Item.ANY)).thenReturn(metsList);
+        when(itemService.getMetadata(rightItem, "relationship", "type", null, Item.ANY)).thenReturn(metsList);
+        when(relationshipDAO.create(any(), any())).thenReturn(relationship);
+        when(relationshipService.find(context,0)).thenReturn(relationship);
+        relationshipService.delete(context, relationship);
+        Mockito.verify(relationshipDAO).delete(context, relationship);
+    }
+
+    @Test
+    public void testUpdate() throws Exception {
+        context.turnOffAuthorisationSystem();
```

Comment:
Again inline comments would be helpful here. There's a lot of content & relationships getting mocked for this test.